lib/state_machine/state_machine.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "consts.hpp"
+#include "state.hpp"
+#include "types.hpp"
+
+#include <optional>
+#include <unordered_map>
+
+namespace hyped::state_machine {
+
+class StateMachine {
+ public:
+  StateMachine() { current_state = State::kIdle; };
+
+  void handleMessage(const Message message);
+
+  std::optional<Message> handleData();
```

Comment:
what is this function supposed to do? it says handle but there isn't any input?

Reply:
unsure on the input at the moment, aim is for this function to handle the messages received from other the other nodes and transition appropriately, will try figure something out for this on Saturday.

Reply:
if no progress has happened with this, add a TODOLater